src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/TaskResult.java

```diff
@@ -70,4 +87,10 @@ public Exception getException() {
         this.shardEndReached = isShardEndReached;
     }
 
+    TaskResult(Exception e, boolean isShardEndReached, List<ChildShard> childShards) {
+        this.exception = e;
+        this.shardEndReached = isShardEndReached;
+        this.childShards = childShards;
```

Comment:
Should we validate that childShards is non-empty if isShardEndReached is true, and null or empty if isShardEndReached is false? Can we add a javadoc for when we would use this constructor?

Also is there a purpose to having the childShards setter above if we can pass childShards into the constructor?

Reply:
We already have the validation step in the data retrieval layer. So we can trust the isShardEndReached here. Yeah, currently we  don't have a use case that we need to set this field after instantiation.